src/api/server.rkt

```diff
@@ -251,6 +308,8 @@
           [else
            (log "Job complete, no timeline, send result.\n")
            (place-channel-put handler (hash-ref completed-work job-id #f))])]
+       [(list 'check job-id handler)
+        (place-channel-put handler (if (hash-has-key? completed-work job-id) job-id #f))]
```

Comment:
Any reason why we return the job-id if it has completed? Why not just true/false? The caller already has to have the job id.